### PR TITLE
Add support for reading EmulationModel results

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR adds interfaces to read emulation model results for simulations. It also fixes some consistency in terminology used in the results interfaces.

We discussed the possibility of adding `len` as an option to decision model simulation results. That is not present here.

### CODE REVIEW

File: src/simulation/hdf_simulation_store.jl [hunk 1]
```diff
@@ -282,27 +297,37 @@ function initialize_problem_storage!(
...
+                col = _make_column_name(name)
+                HDF5.write_dataset(group, col, string.(reqs["columns"]))
+                column_dataset = group[col]
+                datasets = getfield(store.em_data, type)
+                datasets[key] = HDF5Dataset(
+                    dataset,
+                    column_dataset,
+                    # TODO: Josè, why is this broken?
+                    #get_resolution(emulation_params),
+                    get_resolution(first(values(params.decision_models_params))),
```
Comment (daniel-thom) [inline src/simulation/hdf_simulation_store.jl:318]: @jd-lara Need help here.

File: src/simulation/hdf_simulation_store.jl [hunk 2]
```diff
@@ -282,27 +297,37 @@ function initialize_problem_storage!(
...
+                    initial_time,
+                )
+                container_key_lookup[encode_key_as_string(key)] = key
+            end
         end
     end
+
+    buf = IOBuffer()
```
Comment (daniel-thom) [inline src/simulation/hdf_simulation_store.jl:326]: Let me know if what is happening here isn't obvious and I'll add an explanation.

**Comment by daniel-thom** [comment]
> c620eba

@claytonpbarrows I'm not sure I totally follow. Do you want to push your commit to my branch? Please feel free. In case we are missing each other and you are referring to SimulationProblemResults, I expect that you should be getting timestamps in all cases already. Let me know if you aren't.

**Comment by claytonpbarrows** [comment]
Yeah. That was poorly conceived comment. I'll make a PR to address once @jd-lara reviews and this is merged.

**Review by claytonpbarrows — APPROVED** [review]

File: src/simulation/hdf_simulation_store.jl (hunk 1 above)
Comment (jd-lara) [reply src/simulation/hdf_simulation_store.jl:318]: This is part of another bug fix. That's next on the list

File: src/simulation/simulation_problem_results.jl [hunk 3]
```diff
@@ -235,7 +193,8 @@ function _deserialize_key(
     results::SimulationProblemResults,
     name::AbstractString,
 )
-    return deserialize_key(results.optimization_container_metadata, name)
+    !haskey(results.values.container_key_lookup, name) && error("$name is not stored")
```
Comment (jd-lara) [inline src/simulation/simulation_problem_results.jl:196]: Should we display the available keys upon error?

**Review by jd-lara — APPROVED** [review]
All the changes look good to me.

**Comment by jd-lara** [comment]
@claytonpbarrows I merged the PR already so I can rebase the large PWL PR on top of this one.

**Pull request merged**